packages/design-system/src/components/N8nSquareButton/SquareButton.vue

```diff
@@ -0,0 +1,131 @@
+<template functional>
+	<button :class="$style.button" @click="(e) => listeners.click && listeners.click(e)">
+		<span :class="$style[$options.methods.getClass(props)]" :style="$options.methods.getStyles(props)" v-text="props.label" />
+	</button>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+export default Vue.extend({
+	name: 'n8n-square-button',
+	props: {
+		bold: {
+			type: Boolean,
+			default: false,
+		},
+		color: {
+			type: String,
+			default: 'background-dark',
+			validator: (value: string): boolean => ['primary', 'background-dark', 'text-dark', 'text-base', 'text-light', 'text-xlight'].includes(value),
+		},
+		label: {
+			type: String,
+		},
+		size: {
+			type: String,
+			default: 'medium',
+			validator: (value: string): boolean =>
+				['mini', 'small', 'medium', 'large', 'xlarge'].indexOf(value) !== -1,
+		},
+	},
+	methods: {
+		getClass(props: {size: string, bold: boolean}) {
+			return `body-${props.size}${props.bold ? '-bold' : '-regular'}`;
+		},
+		getStyles(props: {color: string}) {
+			const styles = {} as any;
+			if (props.color) {
+				styles.color = `var(--color-${props.color})`;
+			}
+			return styles;
+		},
+	}
+});
+</script>
+
+<style lang="scss" module>
+.button {
+	width: 28px;
+	height: 29px;
+	border-radius: 4px;
+	border: var(--color-background-xlight);
+	cursor: pointer;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+
+	&:hover {
+		.body-medium-bold {
+			color: var(--color-primary)!important;
+		}
+	}
+}
+
+.bold {
+	font-weight: var(--font-weight-bold);
+}
+
+.regular {
+	font-weight: var(--font-weight-regular);
+}
+
+.body-xlarge {
+	font-size: var(--font-size-xl);
+	line-height: var(--font-line-height-xloose);
+}
+
+.body-xlarge-regular {
+	composes: regular;
+	composes: body-xlarge;
+}
+
+.body-xlarge-bold {
+	composes: bold;
+	composes: body-xlarge;
+}
+
+.body-large {
+	font-size: var(--font-size-m);
+	line-height: var(--font-line-height-xloose);
```

Comment:
I am sorry... that's my bad here that I was not more clear. it did not need to be this complicated. you did not have to do all of this. you only need to support your use case...

Reply:
Simplified